src/main/cpp/sim/Simulator.h

```diff
@@ -55,6 +55,14 @@ class Simulator
 	/// Gets the simulator's configuration.
 	SingleSimulationConfig GetConfiguration() const { return m_config; }
 
+	/// Gets the simulator's date
+	unsigned int GetDate() const;
+
+	std::vector<std::vector<Cluster>> GetClusters() const
```

Comment:
This will *copy* **all the clusters** from the entire `Simulator`. There are ways to work around this. I'd suggest replacing the `m_households`, `m_school_clusters`, etc, by a single `struct` that contains `std::vector<Cluster>` values for every type of cluster. You can then make `GetClusters` return a `const&` to that `struct` type.